addons/mail/static/src/discuss/gif_picker/messaging_service.js

```diff
@@ -0,0 +1,14 @@
+/* @odoo-module */
+
+import { Messaging } from "@mail/core/messaging_service";
+import { patch } from "@web/core/utils/patch";
+
+/** @type {Messaging} */
+const messagingPatch = {
+    initMessagingCallback(data) {
+        this._super(data);
+        this.store.hasTenorGifFeature = data.hasTenorGifFeature;
+    },
+};
+
+patch(Messaging.prototype, "GifPicker", messagingPatch);
```

Comment:
And same remark as a previous comment: do `patch()` inline, so the 1st line we clearly see it's a patch.
And the file should be named accordingly, with `_patch` suffix